crates/sui-archival/src/writer.rs

```diff
@@ -236,6 +235,17 @@ impl CheckpointWriter {
         }
         Ok(())
     }
+    fn compress(&self, source: &Path) -> Result<()> {
+        if self.file_compression == FileCompression::None {
+            return Ok(());
+        }
+        let mut input = File::open(source)?;
+        let tmp_file_name = source.with_extension("tmp");
+        let mut output = File::create(&tmp_file_name)?;
+        write(&mut input, &mut output)?;
+        fs::rename(tmp_file_name, source)?;
```

Comment:
i'm a bit puzzled by this function - where is the compression happening?

Reply:
it's happening in the `write` where input in compressed (if enum is set to zstd) and written to output. I am also not happy with the way I named this function, perhaps `compress_and_copy` is a better function name to replace `write`? I'm happy to rename it to a better name

Reply:
renamed to `compress`